Public/Src/Cache/MemoizationStore/Library/Sessions/LocalCache.cs

```diff
@@ -125,20 +127,39 @@ public class LocalCache : OneLevelCache
                         (uint)localCacheConfiguration.RetryCount,
                         scenario: localCacheConfiguration.ScenarioName)
                     : new FileSystemContentStore(
-                        fileSystem, 
+                        fileSystem,
                         clock,
                         rootPath,
                         configurationModel: configurationModel,
                         settings: contentStoreSettings),
-                () => new SQLiteMemoizationStore(
-                    logger,
-                    clock ?? SystemClock.Instance,
-                    memoConfig),
+                CreateMemoizationStoreFactory(logger, clock, memoConfig),
                 PersistentId.Load(fileSystem, rootPath / IdFileName))
         {
             _fileSystem = fileSystem;
         }
 
+        private static Func<IMemoizationStore> CreateMemoizationStoreFactory(ILogger logger, IClock clock, MemoizationStoreConfiguration config)
+        {
+            if (config is SQLiteMemoizationStoreConfiguration sqliteConfig)
+            {
+                return () => new SQLiteMemoizationStore(
+                    logger,
+                    clock ?? SystemClock.Instance,
+                    sqliteConfig);
+            }
+            else if (config is RocksDbMemoizationStoreConfiguration rocksDbConfig)
+            {
+                return () => new RocksDbMemoizationStore(
+                    logger,
+                    clock ?? SystemClock.Instance,
+                    rocksDbConfig);
+            }
+            else
+            {
+                throw new NotSupportedException($"Configuration type for memoization store is unhandled.");

```

Comment:
Minor: Whenever you are constructing an error message, include all the information you reasonably can the first time. Because I guarantee when you see that error message in a log and are trying to figure out what's going on, you will end up going back and putting in more info to debug. In this case, that means identify the type name (or nullity) of config.